### ISSUE TITLE
Big explosion crash on 0.6 update

### ISSUE THREAD

**Issue by Blacksky649** [Majrusz/MajruszsProgressiveDifficultyMod#51]
This new feature seems to crash in my world.

(thanks for the work btw, love where this mod is going)

[crash-2021-05-20_13.11.22-server.txt](https://github.com/Majrusz/MajruszsProgressiveDifficultyMod/files/6517679/crash-2021-05-20_13.11.22-server.txt)

**Comment by The-Bernician-Lamb** [comment]
I'm also getting this issue. I've found that disabling explosions creating fire seems to fix this. So it seems to be related to that part of the explosion feature specifically. 

Edit: My crash report that was generated: https://gist.github.com/NJL5390/716b870a15bffc2100db85e190ec3595

I was able to reproduce this issue by setting off some TNT. It took me a while to reproduce the issue but the TNT block that triggered the crash had another TNT block above it if that helps at all.

Also thought I'd throw in my latest log in case that helps: https://gist.github.com/NJL5390/a85c1853c69dedd9c280707a8cc8e563

Edit 2: This is much easier to reproduce with a TNT Minecart, never fails to make it crash.
